### PR TITLE
Us4 mf

### PR DESCRIPTION
**Pull request by foleymichelle9**
User Story 4 

Added merchant_id dashboard 

Dashboard can only be seen in the nav bar by merchant_employee 

I think there's some cleaning up to with how the files are named, some are called merchant(singular) and some are called merchant_employee. I was a bit confused by what the user stories say and what was already in the files. Let me know which you think we should go with and I will change them.

### CODE REVIEW

File: db/migrate/20200528214738_change_zip_to_be_integer_in_users.rb [hunk 1]
```diff
@@ -1,5 +1,9 @@
 class ChangeZipToBeIntegerInUsers < ActiveRecord::Migration[5.1]
   def change
+<<<<<<< HEAD
+    change_column :users, :zip, 'integer USING CAST (zip AS integer)'
+=======
     change_column :users, :zip, 'integer USING CAST(zip AS integer)'
+>>>>>>> master
```
Comment (foleymichelle9) [inline db/migrate/20200528214738_change_zip_to_be_integer_in_users.rb:7]: Woops these can be deleted

File: app/views/layouts/application.html.erb [hunk 2]
```diff
@@ -15,12 +15,13 @@
       <%= link_to "All Merchants", "/merchants"%>
       <%= link_to "All Items", "/items"%>
       <%= link_to "Cart: #{cart.total_items}", "/cart" %>
-      <%= link_to  "Register", "/register" %>
+      <%= link_to  "Register", "/register" if !current_user %>
```
Comment (foleymichelle9) [inline app/views/layouts/application.html.erb:18]: Added this, not sure which user story it was for but added it while I was here

File: app/controllers/admin/dashboard_controller.rb [hunk 3]
```diff
@@ -1,4 +0,0 @@
-class Admin::DashboardController < ApplicationController
-  def index
-  end 
-end
```
Comment (foleymichelle9) [inline app/controllers/admin/dashboard_controller.rb:4]: Not sure how this got deleted, fixing now

File: app/controllers/sessions_controller.rb [hunk 4]
```diff
@@ -33,7 +33,7 @@ def redirect_user
       user = User.find(session[:user_id])
 
       redirect_to ("/profile") if user.role == "regular"
-      redirect_to ("/merchant") if user.role == "merchant"
+      redirect_to ("/merchant/dashboard") if user.role == "merchant"
```
Comment (perryr16) [inline app/controllers/sessions_controller.rb:36]: thanks for cleaning this up for me!

File: app/views/layouts/application.html.erb (hunk 2 above)
Comment (perryr16) [reply app/views/layouts/application.html.erb:18]: Cool, if its in the story I'm in. I suppose you should only be able to register if you aren't logged in

File: spec/features/merchant_employee/merchant_employee_spec.rb [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+      expect(@merchant1.merchant?).to be_truthy
+
+      visit '/login'
+
+      within("#login-form")do
+        fill_in :email, with: @merchant1.email
+        fill_in :password, with: @password
+        click_button "Login"
+      end
+      click_link "Dashboard"
```
Comment (perryr16) [inline spec/features/merchant_employee/merchant_employee_spec.rb:26]: I'd wrap this in a section called `id="merchant-dashboard"` then make sure a regular user or an admin cant see it.

**Pull request merged**